core/services/pg/event_broadcaster.go

```diff
@@ -80,6 +80,11 @@ func (b *eventBroadcaster) Start(context.Context) error {
 		}
 		b.db = db
 		b.listener = pq.NewListener(b.uri, b.minReconnectInterval, b.maxReconnectDuration, func(ev pq.ListenerEventType, err error) {
+			// sanity check since these can still be called after closing the listener
+			select {
+			case <-b.chStop:
```

Comment:
This will just empty chStop if not empty, otherwise just no-op?
Otherwise I did not get what does it "check" for?